lib/jxl/headers.cc

```diff
@@ -58,7 +58,7 @@ size_t SizeHeader::xsize() const {
 }
 
 Status SizeHeader::Set(size_t xsize64, size_t ysize64) {
-  if (xsize64 > 0xFFFFFFFFull || ysize64 > 0xFFFFFFFFull) {
+  if (xsize64 >= 0xFFFFFFFFull || ysize64 >= 0xFFFFFFFFull) {
```

Comment:
You can safely change these conditions to
```suggestion
  if (xsize64 > 0x40000000ull || ysize64 > 0x40000000ull) {
```
since there is no valid way in which the dimensions can be larger than that.